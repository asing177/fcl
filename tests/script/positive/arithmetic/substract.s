// Subtraction should work the same as addition

global decimal<2> m = 0.01;
global decimal<3> n = 0.001;
global decimal<3> x;
global decimal<3> y;

@initial
init() {
  x = m - n;
  // assert x == 0.009;
  y = n - m;
  // assert y == -0.009;
  transitionTo(@terminal)
}
