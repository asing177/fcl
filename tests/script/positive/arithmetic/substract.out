 
Method arguments and effects:
init: () -> {read m, read n, write x, write y}